packages/server/events-processing-platform/notifications/postmark.go

```diff
@@ -43,18 +46,43 @@ type PostmarkEmailAttachment struct {
 }
 
 type PostmarkProvider struct {
-	Client *postmark.Client
-	log    logger.Logger
+	Client     *postmark.Client
+	log        logger.Logger
+	repository *repository.Repositories
 }
 
-func NewPostmarkProvider(log logger.Logger, serverToken string) *PostmarkProvider {
+func NewPostmarkProvider(log logger.Logger, repo *repository.Repositories) *PostmarkProvider {
 	return &PostmarkProvider{
-		Client: postmark.NewClient(serverToken, ""),
-		log:    log,
+		Client:     postmark.NewClient("", ""),
+		log:        log,
+		repository: repo,
 	}
 }
 
-func (np *PostmarkProvider) SendNotification(ctx context.Context, postmarkEmail PostmarkEmail, span opentracing.Span) error {
+func (np *PostmarkProvider) getClientToken(tenant string) (string, error) {
+	p := np.repository.CommonRepositories.PostmarkApiKeyRepository.GetPostmarkApiKey(tenant)
+	if p.Error != nil {
+		return "", p.Error
+	}
+
+	if p.Result == nil {
+		return "", errors.New("postmark api key not found")
+	}
+
+	serverToken := p.Result.(entity.PostmarkApiKey).Key
+	return serverToken, nil
```

Comment:
return the postmark.NewClient here and rename the function to: getPostmarkClient